### DISCUSSION TITLE
Inconsistent `tags` argument in `Taggable.{__init__, _with_new_tags, tagged}`

### DISCUSSION THREAD

**Discussion by matthiasdiener** [inducer/pytools#137] [Q&A] [answered]
I wonder whether they should all accept `tags: ToTagSetConvertible` instead of the current mix of `tags: ToTagSetConvertible` and  `tags: FrozenSet[Tag]`.

https://github.com/inducer/pytools/blob/ea121e15354a23229dfb0c44c9b67592de3889ce/pytools/tag.py#L253-L284

**Comment by inducer** [comment, marked as answer]
That's on purpose. Constructors are supposed to be cheap. `ToTagSetConvertible` is subject to normalization, after which it becomes `FrozenSet[Tag]`. Normalization is kept out of the constructor code path specifically to avoid paying the normalization cost repeatedly.

**Reply by matthiasdiener** [reply]
Ok, I see.
